React/Base/RCTConvert.m

```diff
@@ -375,6 +375,38 @@ + (NSLocale *)NSLocale:(id)json
     NSWritingDirectionNatural,
     integerValue)
 
++ (NSLineBreakStrategy)NSLineBreakStrategy:(id)json RCT_DYNAMIC
+{
+  static NSDictionary *mapping;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    if (@available(iOS 14.0, *)) {
+      mapping = @{
+        @"none" : @(NSLineBreakStrategyNone),
+#if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 140000
+        @"standard" : @(NSLineBreakStrategyStandard),
+        @"hangul-word" : @(NSLineBreakStrategyHangulWordPriority),
+        @"push-out": @(NSLineBreakStrategyPushOut)
+#else
+        @"standard" : @(NSLineBreakStrategyNone),
+        @"hangul-word" : @(NSLineBreakStrategyNone),
+        @"push-out": @(NSLineBreakStrategyNone)
+#endif
+      };
+    } else {
+      mapping = @{
+        @"none" : @(NSLineBreakStrategyNone),
+        @"standard" : @(NSLineBreakStrategyNone),
+        @"hangul-word" : @(NSLineBreakStrategyNone),
+        @"push-out": @(NSLineBreakStrategyNone)
+      };
```

Comment:
From @javache:
I'd avoid adding this mapping just for the purpose of enum validation.

Reply:
<img width="972" alt="image" src="https://user-images.githubusercontent.com/26326015/193147142-6edc24da-9c13-42e9-8290-659248da5f2b.png">

Could it be empty like the screenshot? Please advise 🙏

Reply:
```diff
* (NSLineBreakStrategy)NSLineBreakStrategy:(id)json RCT_DYNAMIC
{
  static NSDictionary *mapping;
  static dispatch_once_t onceToken;
  dispatch_once(&onceToken, ^{
    if (@available(iOS 14.0, *)) {
      mapping = @{
        @"none" : @(NSLineBreakStrategyNone),
#if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 140000
        @"standard" : @(NSLineBreakStrategyStandard),
        @"hangul-word" : @(NSLineBreakStrategyHangulWordPriority),
        @"push-out": @(NSLineBreakStrategyPushOut)
+#endif
-#else
-        @"standard" : @(NSLineBreakStrategyNone),
-        @"hangul-word" : @(NSLineBreakStrategyNone),
-        @"push-out": @(NSLineBreakStrategyNone)
-#endif
      };
    } else {
      mapping = @{
-        @"none" : @(NSLineBreakStrategyNone),
-        @"standard" : @(NSLineBreakStrategyNone),
-        @"hangul-word" : @(NSLineBreakStrategyNone),
-        @"push-out": @(NSLineBreakStrategyNone)
      };
    }
  });

  NSLineBreakStrategy type = RCTConvertEnumValue("NSLineBreakStrategy", mapping, @(NSLineBreakStrategyNone), json).integerValue;
  return type;
}
```

Reply:
Do we even need the `#if` define check here? Apple pushes SDK upgrades quite aggressively, so I assume we can freely use bits from the iOS14 SDK at this point. 

Could you restructure this as

```
if (@available(iOS 14.0, *)) {
   dispatch_once(...);
   return RCTConvertEnumValue(...)
} else {
  return NSLineBreakStrategyNone;
}
```

Reply:
I agree with you, I changed it as below.

```objc
+ (NSLineBreakStrategy)NSLineBreakStrategy:(id)json RCT_DYNAMIC
{
    if (@available(iOS 14.0, *)) {
      static NSDictionary *mapping;
      static dispatch_once_t onceToken;
      dispatch_once(&onceToken, ^{
        mapping = @{
          @"none" : @(NSLineBreakStrategyNone),
          @"standard" : @(NSLineBreakStrategyStandard),
          @"hangul-word" : @(NSLineBreakStrategyHangulWordPriority),
          @"push-out": @(NSLineBreakStrategyPushOut)
        };
      });
      return RCTConvertEnumValue("NSLineBreakStrategy", mapping, @(NSLineBreakStrategyNone), json).integerValue;
    } else {
      return NSLineBreakStrategyNone;
    }
}
```